src/highdicom/sr/content.py

```diff
@@ -524,6 +744,120 @@ def __init__(
             )
         self.append(source_image)
 
+    @classmethod
+    def from_seg_dataset(
+            cls,
+            dataset: Dataset,
+            frame_number: int
+        ) -> 'ReferencedSegmentationFrame':
+        """Construct the content item directly from a segmentation dataset
+
+        Parameters
+        ----------
+        dataset: pydicom.dataset.Dataset
+            Dataset representing a segmentation containing the referenced
+            segment.
+        frame_number: int
+            Number of the frame to which the reference applies. Note that
+            one-based indexing is used to index the frames.
+
+        Returns
+        -------
+        ReferencedSegment
+            Content item representing a reference to the segment
+
+        Notes
+        -----
+        This method will attempt to deduce source image information
+        from information provided in the segmentation instance. If
+        available, it will use information specific to the segment
+        and frame numbers (if any) provided using the Derivation
+        Image Sequence item for the given frame. If this information
+        is not present in the segmentation dataset, it will instead
+        use the information in the Referenced Series Sequence, which
+        applies to all segments and frames present in the segmentation
+        instance.
+
```

Comment:
Please add a `Raises` section to indicate under what circumstances a `ValueError` will be raised

Reply:
Resolved in 4cc5f0e